src/algebra/algebra/subalgebra.lean

```diff
@@ -432,10 +432,26 @@ end algebra
 namespace subalgebra
 open algebra
 
-variables {R : Type u} {A : Type v}
-variables [comm_semiring R] [semiring A] [algebra R A]
+variables {R : Type u} {A : Type v} {B : Type w}
+variables [comm_semiring R] [semiring A] [algebra R A] [semiring B] [algebra R B]
 variables (S : subalgebra R A)
 
+instance [subsingleton A] : subsingleton (subalgebra R A) :=
+⟨λ B C, ext (λ x, by { simp only [subsingleton.elim x 0, zero_mem] })⟩
+
+instance [subsingleton (subalgebra R A)] : subsingleton (A →ₐ[R] B) :=
+⟨λ f g, alg_hom.ext $ λ a,
+  have a ∈ (⊥ : subalgebra R A) := subsingleton.elim (⊤ : subalgebra R A) ⊥ ▸ mem_top,
+  let ⟨x, hx⟩ := set.mem_range.mp (mem_bot.mp this) in
+  hx ▸ (f.commutes _).trans (g.commutes _).symm⟩
+
+instance alg_equiv.subsingleton_left [subsingleton (subalgebra R A)] : subsingleton (A ≃ₐ[R] B) :=
+⟨λ f g, alg_equiv.ext (λ x, alg_hom.ext_iff.mp (subsingleton.elim f.to_alg_hom g.to_alg_hom) x)⟩
+
+instance alg_equiv.subsingleton_right [subsingleton (subalgebra R B)] : subsingleton (A ≃ₐ[R] B) :=
+⟨λ f g, eq.trans (alg_equiv.symm_symm.symm)
+  (by rw [subsingleton.elim f.symm g.symm, alg_equiv.symm_symm])⟩
```

Comment:
So for now, maybe change `instance` to `lemma` and add a comment like `-- TODO[gh-6025]: make this an instance once safe to do so` (xref #gh-6025)